### PR TITLE
distinguish between in-cluster and external cluster for spoke cluster

### PR DESCRIPTION
**Pull request by skeeey**

### CODE REVIEW

File: pkg/agent/agent.go [hunk 1]
```diff
@@ -72,14 +72,16 @@ func init() {
...
 
 	StatusSyncInterval                     time.Duration
 	AppliedManifestWorkEvictionGracePeriod time.Duration
 
 	Burst int
 	QPS   float32
+
+	KubeConfig string
+
+	eventRecorder events.Recorder
```
Comment (clyang82) [inline pkg/agent/agent.go:84]: can we capitalize `eventRecorder` so that it can be used outside?

Comment (skeeey) [reply pkg/agent/agent.go:84]: I don't think we should use this outside, this recorder has component name, it should only use for the this agent

**Review by clyang82 — APPROVED** [review]
LGTM

**Pull request merged**